docs/recipes/typescript.md

```diff
@@ -0,0 +1,56 @@
+# TypeScript
+
+AVA comes bundled with a TypeScript definition file. This allows developers to leverage TypeScript for writing tests.
+
+## Setup
+
+First install the TypeScript compiler [tsc](https://github.com/Microsoft/TypeScript).
+
+```
+$ npm install tsc --save-dev
+```
+
+Create a [`tsconfig.json`](https://github.com/Microsoft/TypeScript/wiki/tsconfig.json) file. This file specifies the compiler options required to compile the project or the test file.
+
+```json
+{
+	"compilerOptions": {
+		"module": "commonjs",
+		"target": "es2015"
```

Comment:
Isn't it quite specific per project? I mean, some people will use TypeScript for the entire project, some might only use it for the tests. I don't want to force them in using a specific approach or directory structure. This recipe is just for the basics, they can always go from here and do some more advanced stuff with it, no?